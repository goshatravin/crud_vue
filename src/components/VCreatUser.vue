<template>
    <div class="home" v-show="toggle">
        <div class="new">
                    <div class="user-profile___btn_close">
            <input class="close__input" type="submit" value="Close" v-on:click='closePopUp'>
        </div>
        <div class="user-profile__avatar">
            <p>Your picture</p>
            <img class="avatar__src" src="#" alt="">
        </div>
        <div class="user-profile__name">
            <p>Your name</p>
            <input class="name__input" type="text" v-model="new_user.name">
        </div>
        <div class="user-profile__position">
            <p>Your position</p>
            <input class="possition__input" type="text" v-model="new_user.position">
        </div>
        <div class="user-profile__address">
            <p>Your address</p>
            <input class="address__input" type="text" v-model="new_user.address">
        </div>
        <div class="user-profile__company">
            <p>Your company</p>
            <input class="company__input" type="text" v-model="new_user.company">
        </div>
        <div class="user-profile___btn_save">
                <input class="save__input" type="submit" value="Save" v-on:click='saveNewUser'>
        </div>
        </div>
        <div class="eddit">
            <input type="text" v-model="edit_user['data'].name">
             <input type="text" v-model="edit_user['data'].position">
              <input type="text" v-model="edit_user['data'].address">
               <input type="text" v-model="edit_user['data'].company">
               <input type="submit" v-on:click='saveEdituser' >
               {{edit_user}}
        </div>
    </div>
</template>

<script>
import { mapState } from 'vuex';

export default {
    name: 'VCreatUser',
    data() {
        return {
            new_user: {
                name: 'George',
                position: 'Ceo',
                address: 'epronovskay 21',
                company: 'BALT',
            },
        };
    },
    computed: {
        ...mapState({
            toggle: (state) => state.newUser.toggle,
            edit_user: (state) => state.newUser.editUser,
        }),
    },
    methods: {
        // Close pop-up with registration form for registration new users
        closePopUp() {
            this.$store.dispatch('TOGGLE_POP_UP', false);
            this.new_user = {};
        },
        // Save new user and close pop-up
        saveNewUser() {
            this.$store.dispatch('SAVE_NEW_USER', this.new_user);
            this.new_user = {};
            this.closePopUp();
        },
        saveEdituser() {
            this.$store.dispatch('SAVE_EDITED_USER', this.edit_user);
        },
    },
};
</script>

<style>

</style>
