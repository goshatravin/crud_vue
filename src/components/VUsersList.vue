<template>
    <div class="home">
        <div class="new-user">
            <button v-on:click='addNewUser'>New user</button>
        </div>
        <div class="all-users">
            <div class="new_one" v-for="(item, index) of usersArray"
            :key='item.name'
            >
                <p>{{item.name}}</p>
                <p>{{item.position}}</p>
                <p>{{item.address}}</p>
                <p>{{item.company}}</p>
                <p :name='index' v-on:click="edit">EDIT</p>
                <p>DEL</p>
            </div>
        </div>
    </div>
</template>

<script>
import { mapState } from 'vuex';

export default {
    name: 'VUsersList',
    methods: {
        // Show pop-up with registration form for registration new users
        addNewUser() {
            this.$store.dispatch('TOGGLE_POP_UP', true);
        },
        // eslint-disable-next-line no-unused-vars
        edit(e, target) {
            const myObj = {
                data: this.usersArray[e.target.getAttribute('name')],
                id: e.target.getAttribute('name'),
            };
            this.$store.dispatch('EDIT_USER', myObj);
        },
    },
    computed: {
        ...mapState({
            usersArray: (state) => state.newUser.usersArray,
        }),
    },
};
</script>

<style>
.new_one{
    border: 1px solid black;
}
</style>
