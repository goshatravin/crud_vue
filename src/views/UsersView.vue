<template>
    <div class="home">
        <VCreatUser/>
        <VUsersList/>
    </div>
</template>

<script>
import VCreatUser from '@/components/VCreatUser.vue';
import VUsersList from '@/components/VUsersList.vue';

export default {
    name: 'UsersView',
    components: {
        VCreatUser,
        VUsersList,
    },
};
</script>

<style>

</style>
