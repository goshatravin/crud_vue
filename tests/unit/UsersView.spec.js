import { shallowMount } from '@vue/test-utils';
import UsersView from '@/views/UsersView.vue';
import VUsersList from '@/components/VUsersList.vue';
import VCreatUser from '@/components/VCreatUser.vue';

describe('UsersView', () => {
    const build = () => {
        const wrapper = shallowMount(UsersView);

        return {
            wrapper,
            usersList: () => wrapper.find(VUsersList),
            creatUser: () => wrapper.find(VCreatUser),
        };
    };

    it('Renders the component', () => {
        // arrange
        const { wrapper } = build();
        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Renders main child components', () => {
        // arrange
        const { usersList, creatUser } = build();
        // assert
        expect(usersList().exists()).toBe(true);
        expect(creatUser().exists()).toBe(true);
    });
});
