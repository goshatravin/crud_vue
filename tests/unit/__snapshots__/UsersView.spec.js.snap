// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersView Renders the component 1`] = `
<div class="home">
  <vcreatuser-stub></vcreatuser-stub>
  <vuserslist-stub></vuserslist-stub>
</div>
`;
